domain/consensus/utils/txscript/engine.go

```diff
@@ -358,9 +361,9 @@ func (vm *Engine) currentScript() []parsedOpcode {
 
 // checkHashTypeEncoding returns whether or not the passed hashtype adheres to
 // the strict encoding requirements if enabled.
-func (vm *Engine) checkHashTypeEncoding(hashType SigHashType) error {
-	sigHashType := hashType & ^SigHashAnyOneCanPay
-	if sigHashType < SigHashAll || sigHashType > SigHashSingle {
+func (vm *Engine) checkHashTypeEncoding(hashType consensushashing.SigHashType) error {
+	sigHashType := hashType & ^consensushashing.SigHashAnyOneCanPay
+	if sigHashType < consensushashing.SigHashAll || sigHashType > consensushashing.SigHashSingle {
```

Comment:
Looks correct according to https://github.com/bitcoin/bitcoin/blob/master/src/script/interpreter.cpp#L193-L194